simplelru/lru.go

```diff
@@ -159,14 +225,56 @@ func (c *LRU[K, V]) Resize(size int) (evicted int) {
 		diff = 0
 	}
 	for i := 0; i < diff; i++ {
-		c.removeOldest()
+		if c.useSieve {
+			c.performSieveEviction()
+		} else {
+			c.removeOldest()
+		}
 	}
+
 	c.size = size
 	return diff
 }
 
+// performSieveEviction - runs a eviction by running Sieve Algorithm and returns the evicted value.
+func (c *LRU[K, V]) performSieveEviction() (key K, value V, ok bool) {
+	c.getSieveCandidate()
+	if c.hand != nil {
+		candidate := c.hand
+		c.hand = c.hand.PrevEntry()
+		c.removeElement(candidate)
+		return candidate.Key, candidate.Value, true
+	}
+
+	return
+}
+
+// getSieveCandidate evicts an entry based on sieve algorithm.
+func (c *LRU[K, V]) getSieveCandidate() {
+	if c.Len() == 0 {
+		return
+	}
+
+	if c.hand == nil {
+		c.hand = c.evictList.Back()
+	}
+
+	for c.hand != nil && c.hand.Visited {
+		c.hand.Visited = false
+		c.hand = c.hand.PrevEntry()
+		if c.hand == nil {
+			c.hand = c.evictList.Back()
+		}
+	}
+}
+
 // removeOldest removes the oldest item from the cache.
 func (c *LRU[K, V]) removeOldest() {
+	if c.useSieve {
+		c.performSieveEviction()
+		return
+	}
+
```

Comment:
better to put them in else so that it would be easier to understand they are LRU part

Reply:
Same as the other comment.

Reply:
oh, I was referring to the part after l276. I think it might be easier to read if they are inside the else branch (so we know it is LRU code)

Reply:
Sorry. Not sure if I understand this correctly.

Are you suggestion something like this?

```
func (c *Cache[K, V]) removeOldest() {
	if c.useSieve {
		c.performSieveEviction()
		return
	} else {
	          if ent := c.evictList.Back(); ent != nil {
		          c.removeElement(ent)
	          }
	}
}
```

Reply:
yup

Reply:
Wouldn't this be the same as this comment (https://github.com/venkatsvpr/golang-lru/pull/1/files/798462d290190cf0008c3bb8606046df184c9eda#r1452870951)

Does this make sense?

Reply:
Sorry I was too lazy today. 
That one has two comments, one is about named returns, the other one is about wrapping the LRU code in else branch. This one is just about the else branch. Does it make sense?

Reply:
Yes - The second part of the comment. Sorry my bad. I should have clarified.

Reply:
Yes, it is easier to read with the else branch (but not sure whether it would be idiomatic Go code though)